tensorflow/core/kernels/fused_batch_norm_op.h

```diff
@@ -43,6 +45,66 @@ struct InvVarianceToVariance {
   void operator()(const Eigen::GpuDevice& d, double epsilon, int sample_size,
                   int channels, T* variance);
 };
+
+
+// Functor used by FusedBatchNormGradOp to do the computations when is_training=False.
+// Both CPU and GPU will use this functor.
+template <typename Device, typename T>
+struct FusedBatchNormFreezeGrad {
+  void operator()(const Device& d, const Tensor& y_backprop_input,
+                  const Tensor& x_input, const Tensor& scale_input,
+                  const Tensor& mean_input, const Tensor& variance_input,
+                  T epsilon, Tensor* x_backprop_output,
+                  Tensor* scale_backprop_output, Tensor* offset_backprop_output,
+                  typename TTypes<T>::Vec scratch1, typename TTypes<T>::Vec scratch2) {
```

Comment:
Are these two scratch allocation needed? Could you follow the Eigen implementation of FusedBatchNorm, where  no temp allocation is needed (you can still use something like "Eigen::Tensor<T, 1, Eigen::RowMajor> mean(depth)" though).

Reply:
Seems like `Eigen::Tensor<T, 1, Eigen::RowMajor> scratch1(depth)` only allocate memory on CPUs? In a GPU kernel, using this ends up with `CUDA_ERROR_ILLEGAL_ADDRESS`.

Reply:
Thanks, sounds good to me.